includes/sanitizers/class-amp-gallery-block-sanitizer.php

```diff
@@ -149,105 +125,17 @@ public function sanitize() {
 				continue;
 			}
 
-			list( $width, $height ) = $this->get_carousel_dimensions( $node );
-
-			$amp_carousel = AMP_DOM_Utils::create_node(
-				$this->dom,
-				'amp-carousel',
-				[
-					'width'  => $width,
-					'height' => $height,
-					'type'   => 'slides',
-					'layout' => 'responsive',
-				]
-			);
-
+			$images_and_captions = [];
 			foreach ( $images as $image ) {
-				$slide = AMP_DOM_Utils::create_node(
-					$this->dom,
-					'div',
-					[ 'class' => 'slide' ]
-				);
-
-				// Ensure the image fills the entire <amp-carousel>, so the possible caption looks right.
-				if ( 'amp-img' === $image->tagName ) {
-					$image->setAttribute( 'layout', 'fill' );
-					$image->setAttribute( 'object-fit', 'cover' );
-				} elseif ( isset( $image->firstChild->tagName ) && 'amp-img' === $image->firstChild->tagName ) {
-					// If the <amp-img> is wrapped in an <a>.
-					$image->firstChild->setAttribute( 'layout', 'fill' );
-					$image->firstChild->setAttribute( 'object-fit', 'cover' );
-				}
-
-				$possible_caption_text = $this->possibly_get_caption_text( $image );
-				$slide->appendChild( $image );
-
-				// Wrap the caption in a <div> and <span>, and append it to the slide.
-				if ( $possible_caption_text ) {
-					$caption_wrapper = AMP_DOM_Utils::create_node(
-						$this->dom,
-						'div',
-						[ 'class' => 'amp-wp-gallery-caption' ]
-					);
-					$caption_span    = AMP_DOM_Utils::create_node( $this->dom, 'span', [] );
-					$text_node       = $this->dom->createTextNode( $possible_caption_text );
-
-					$caption_span->appendChild( $text_node );
-					$caption_wrapper->appendChild( $caption_span );
-					$slide->appendChild( $caption_wrapper );
-				}
-
-				$amp_carousel->appendChild( $slide );
```

Comment:
This 'red' section in the diff is moved to `AMP_Carousel::create_and_get()`, with few changes.